streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -166,6 +248,11 @@ public String metricsScope() {
      * This store supplier can be passed into a {@link #windowStoreBuilder(WindowBytesStoreSupplier, Serde, Serde)}.
      * If you want to create a {@link TimestampedWindowStore} you should use
      * {@link #persistentTimestampedWindowStore(String, Duration, Duration, boolean)} to create a store supplier instead.
+     * <p>
+     * Note that it is not safe to change the value of {@code retentionPeriod} between
```

Comment:
I worry slightly that adding this warning about not changing retention period between restarts will give the false impression that it is OK to change, for example, window size between restarts. While I don't think that's a good idea, I also don't think anything necessarily breaks if a user does that (unlike for history retention where something definitely could break, if segment interval is changed under the hood as well) so it's unclear whether that should be discouraged in the docs or not.

Reply:
I think we can give it a try as you propose -- docs are "easy" to change, so I would not worry about it too much.